.dependency_decisions.yml

```diff
@@ -251,3 +251,27 @@
     :why: Unlicense
     :versions: []
     :when: 2020-11-07 22:16:03.105598000 Z
+- - :permit
+  - Mozilla Public License 2.0
+  - :who: Charles546
+    :why: Needed to support vault
+    :versions: []
+    :when: 2025-01-06 19:45:34.965440000 Z
+- - :approve
+  - github.com/hashicorp/go-retryablehttp
+  - :who: Charles546
+    :why: Needed to support vault
+    :versions: []
+    :when: 2025-01-06 19:51:25.738123000 Z
+- - :approve
+  - github.com/hashicorp/vault/api
+  - :who: Charles546
+    :why: Needed to support vault
+    :versions: []
+    :when: 2025-01-06 19:51:46.891654000 Z
+- - :approve
+  - github.com/hashicorp/vault/api/auth/kubernetes
+  - :who: Charles546
+    :why: Needed to support vault
+    :versions: []
+    :when: 2025-01-06 19:52:12.119818000 Z
```

Comment:
This is probably wrong. Should not be releasing MIT code with MPLv2 components. Should I release this in a separate repo with MPLv2 license?

Reply:
Seems like the main vault component is under their new closed source license, but the subcomponent you're using is MPL2.

Looks like the OpenBao fork is also MPL. https://github.com/openbao/openbao/tree/main, as is https://github.com/hashicorp/vault-client-go

Making a whole separate project / plugin doesn't seem ideal... maybe you can just make the API calls you need directly? Or https://github.com/mittwald/vaultgo is MIT licensed if that looks good security-wise and can do what you need?

Reply:
vaultgo uses almost the same set of Hashcorp MPLv2 libraries. If they can release it under MIT. Why can't I?

Reply:
😬 maybe they don't have good legal advice? 😅

Reply:
https://softwareengineering.stackexchange.com/questions/317944/can-i-include-code-licensed-under-mpl-within-a-project-under-mit-license
https://www.mozilla.org/en-US/MPL/2.0/FAQ/

So sounds like it's Ok, especially since you're not modifying (or even vendoring) the source code. That said, it may be wise to consult an actual lawyer at some point.

Reply:
Yes, the Q8 from the FAQ seems matching the use case here.